test/e2e/app-dir/next-image/next-image.test.ts

```diff
@@ -11,6 +13,34 @@ describe('app dir - next-image', () => {
   }
 
   describe('ssr content', () => {
+    it('should handle HEAD requests for uncached images', async () => {
+      const imagesDir = join(next.testDir, '.next/cache/images')
+      await fs.remove(imagesDir).catch(() => {})
+
+      const $ = await next.render$('/')
+      const imageUrl = $('#app-layout').attr('src')
+
+      const headRes = await next.fetch(imageUrl, { method: 'HEAD' })
+      expect(headRes.status).toBe(200)
+      expect(headRes.headers.get('content-type')).toMatch(/^image\//)
+      expect(headRes.headers.get('X-Nextjs-Cache')).toBe('MISS')
+
+      const contentLength = headRes.headers.get('content-length')
+      expect(Number(contentLength || '0')).toBeGreaterThan(0)
```

Comment:
This should also assert that the body is empty here

```suggestion
      expect(Number(contentLength || '0')).toBeGreaterThan(0)
      const headBody = await getRes.arrayBuffer()
      expect(headBody.byteLength).toBe(0)
```